### DISCUSSION TITLE
GAME MECHANIC: Looking for input on "rules of engagement" (or auto-attack)

### DISCUSSION THREAD

**Discussion by Volte6** [GoMudEngine/GoMud#104] [General]
When should a players character automatically begin attacking mobs, if at all? Here are some scenarios to consider:

* They are in a party, and the party leader attacks a mob
* They are in a party and a non-party leader attacks a mob
* A mob hits the player directly (engages in a 1-way attack on the player)
* Similar to above, a mob aggros the player directly (before a hit is landed)
* A harmful spell is cast on the player by a mob
* The player defeats one of several aggro'd mobs (i.e. there are more still attacking)

There are probably some other scenarios too.

Should players only engage in combat with mobs when they explicitly indicate they wish to? Running through an area where mobs hit the player would become difficult if you constantly auto-engaged in combat.

**Comment by ArmoredArtist** [comment]
Some ideas - I would set a flag to auto attack back when attacked in any way (on or off) that way when traveling you could just flag it off to do so. I think if your in a group the party leader should have to initiate the attack always then you auto attack if he does first (by grouping up your agreeing to fight together). You do not want your personal attack flag logic competing with anyone in your group because the leader might not swing while traveling but you have it set to on and start attacking by yourself so it should default to the leaders combat logic.

**Comment by Volte6** [comment]
Well having a flag or configurable behavior is fine, but there should still be some default behavior prior to players overriding it...

**Reply by ArmoredArtist** [reply]
Maybe default on the side of caution to full auto attack back to everything. When they travel and it annoys them then they will figure out how to flag it off :)

**Reply by AngstPunk** [reply]
I've seen muds that use an 'assist -player-' command to assist the group member in combat. They also had a system where the stance of the player mattered. For example. We are in a party, and an aggressive mob attacks you. I assist. My stance is aggressive, yours is normal. The monster turns focus to me. However, if your damage dealt is more, the monster does not lose focus on you. This allowed a round robin effect of players wasting a combat round by entering a 'normal' or 'wary' stance to switch aggression to another player, but without leaving combat.

**Reply by Volte6** [reply]
"parties" have a config to auto-assist (on or off), so that's partially covered.

There is also frontrank/backrank, I think.

We have a lot of systems we are attempting to balance, so the more ownership someone can take in:

1. Designing a good system
2. Designing a flexible/extensible/configurable system

The better we'll do in terms of locking down how we want the default behaviors to be.

**Reply by AngstPunk** [reply]
I can take a look at this if you want. I haven't gone through the code enough to even know where it is yet, but I've dealt with it before, so I can give it a try. I can look at the way it is coded now and see what might need to happen. I already see a command like 'assist' or 'rescue' being added, and a way to quicker determine party, stance and wary, but it doesn't seem that hard.

**Reply by Volte6** [reply]
There is some work being done on refactoring the combat system altogether, so I would hold back on this just now... however, figuring out what the rules should be and writing them out with any edge cases etc. would be a good start... would also help create discussion around it.
